src/main/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/validation/TransactionIdValidationInterceptor.java

```diff
@@ -0,0 +1,41 @@
+package uk.gov.companieshouse.confirmationstatementapi.interceptor.validation;
+
+import org.apache.commons.lang.StringUtils;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.servlet.HandlerMapping;
+import uk.gov.companieshouse.confirmationstatementapi.utils.ApiLogger;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Map;
+
+import static uk.gov.companieshouse.confirmationstatementapi.utils.Constants.TRANSACTION_ID_KEY;
+
+@Component
+public class TransactionIdValidationInterceptor implements HandlerInterceptor {
+
+    private static final int MAX_LENGTH = 50;
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
+        final Map<String, String> pathVariables = (Map<String, String>) request
+                .getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE);
+        final var transactionId = pathVariables.get(TRANSACTION_ID_KEY);
+
+        if (StringUtils.isBlank(transactionId)) {
+            ApiLogger.debug("No transaction URL id supplied");
```

Comment:
See previous comments about SubmissionId validation with regard to logging and return code (400), should be applied in this class also.